pkg/piecestore/psserver/agreementsender/agreementsender.go

```diff
@@ -102,7 +102,7 @@ func (as *AgreementSender) SendAgreementsToSatellite(ctx context.Context, satID
 			}
 		}
 		// Delete from PSDB by signature
-		if err = as.DB.DeleteBandwidthAllocationBySignature(agreement.Signature); err != nil {
+		if err = as.DB.DeleteBandwidthAllocationBySerialnum(rba.PayerAllocation.SerialNumber); err != nil {
```

Comment:
Have we tested / analyzed that this actually solves the JIRA ticket, or are we just assuming it does?

Reply:
yes I have tested it. I made the agreement sender time to 5sec. Ie the agreement sender run every 5sec instead of 1hr. If everything works , the bwagreement-table should be empty without any entries, and saw the result as expected. The bandwidth_agrement table is empty. 

Also while agreements are being sent, I uploaded about 500MB file to see if it exits out.. or any errors happen. Expected out is nothing should happen and the bandwidth_agreement table should be empty ie all the agreeements are sent to satellite. Result was as expected.